tests/src/testfebases.cpp

```diff
@@ -0,0 +1,108 @@
+}
+
+template <typename TestSuiteType, typename... Bases>
+void checkBases(TestSuiteType& t, const std::vector<int>& expectedNumberOfChildren,
+                const std::vector<int>& expectedSize, const std::string& basisName, const Bases&... bases) {
+  assert(sizeof...(bases) == expectedNumberOfChildren.size() && "Input size mismatch for number of children.");
+  assert(sizeof...(bases) == expectedSize.size() && "Input size mismatch for local view size.");
+  int i = 0;
+  (
+      [&] {
+        checkScalars(t, bases.numberOfChildren(), expectedNumberOfChildren[i],
+                     " Number of children is incorrect for i = " + std::to_string(i) + basisName);
+        checkScalars(t, static_cast<int>(bases.size()), expectedSize[i],
+                     " Size is incorrect for i = " + std::to_string(i) + basisName);
+        std::vector<typename std::remove_cvref_t<decltype(bases.localView())>::MultiIndex> dofs;
+        bases.globalFlatIndices(dofs);
+        checkScalars(t, dofs.size(), bases.size(),
+                     " Size of dofs and args are not equal for i = " + std::to_string(i) + basisName);
+        ++i;
+      }(),
+      ...);
+}
+
+auto FEBasesTest() {
+  TestSuite t("FEBasesTest");
+  using Grid = Dune::YaspGrid<2>;
+
+  Dune::FieldVector<double, 2> bbox       = {4, 2};
+  std::array<int, 2> elementsPerDirection = {1, 1};
+  const auto grid                         = std::make_shared<Grid>(bbox, elementsPerDirection);
+
+  const auto gridView = grid->leafGridView();
+
+  using namespace Dune::Functions::BasisFactory;
+
+  const auto firstOrderLagrangePreBasis  = lagrange<1>();
+  const auto secondOrderLagrangePreBasis = lagrange<2>();
+
+  const auto firstOrderLagrangePowerPreBasis  = power<2>(firstOrderLagrangePreBasis);
+  const auto secondOrderLagrangePowerPreBasis = power<5>(secondOrderLagrangePreBasis);
+
+  const auto scalarScalarCompositePreBasis = composite(firstOrderLagrangePreBasis, secondOrderLagrangePreBasis);
+  const auto scalarPowerCompositePreBasis  = composite(firstOrderLagrangePreBasis, secondOrderLagrangePowerPreBasis);
+
+  const auto powerScalarCompositePreBasis = composite(firstOrderLagrangePowerPreBasis, secondOrderLagrangePreBasis);
+  const auto powerPowerCompositePreBasis = composite(firstOrderLagrangePowerPreBasis, secondOrderLagrangePowerPreBasis);
+
+  const auto scalarCompositePreBasis = composite(firstOrderLagrangePreBasis);
+  const auto powerCompositePreBasis  = composite(firstOrderLagrangePowerPreBasis);
+
+  /// Types of scalar FE bases
+  const auto scalar1 = getBasis(gridView, firstOrderLagrangePreBasis);
+  const auto scalar2 = getBasis(gridView, secondOrderLagrangePreBasis);
+
+  /// Types of power FE bases
+  const auto power1 = getBasis(gridView, firstOrderLagrangePowerPreBasis);
+  const auto power2 = getBasis(gridView, secondOrderLagrangePowerPreBasis);
+
+  /// Types of composite FE bases
+  const auto composite1 = getBasis(gridView, scalarScalarCompositePreBasis);
```

Comment:
Lots of test cases here. Good!

 I would suggest we add a few nested ones. `power(composite(composite(...)), maybe from https://gitlab.dune-project.org/staging/dune-functions/-/blob/master/dune/functions/functionspacebases/test/makebasistest.cc?ref_type=heads and fix the code. Then we never have to touc hit again.

Additionall we should add one test with vector-valued fe such as nedelc. To really have the correct forever.